tracer/src/Datadog.Trace/DataStreamsMonitoring/DataStreamsManager.cs

```diff
@@ -124,7 +131,8 @@ public void InjectPathwayContext<TCarrier>(PathwayContext? context, TCarrier hea
                 previousContext = PreviousCheckpoint.Value.Context;
             }
 
-            var edgeStartNs = DateTimeOffset.UtcNow.ToUnixTimeNanoseconds();
+            var nowNs = DateTimeOffset.UtcNow.ToUnixTimeNanoseconds();
+            var edgeStartNs = previousContext == null && timeInQueueMs > 0 ? nowNs - (timeInQueueMs * 1_000_000) : nowNs;
```

Comment:
When I mentioned about extra tests, this was the logic I was talking about, as it's not very intuitive (I believe you that its right, but I don't really understand _why_) 😄

Reply:
Yes, I'll push tests to verify this behavior later today.

Reply:
Cool, thanks! Wasn't trying to be pushy, just saw you'd already addressed pretty much everything else 😄